src/SSVOpenHexagon/Core/HexagonGame.cpp

```diff
@@ -288,10 +287,12 @@ void HexagonGame::newGame(const std::string& mPackId, const std::string& mId,
 
     lua = Lua::LuaContext{};
     initLua();
-
     runLuaFile(levelData->luaScriptPath);
     runLuaFunction<void>("onInit");
     runLuaFunction<void>("onLoad");
+
+    player = CPlayer{ssvs::zeroVec2f, getSwapCooldown(), &levelStatus.playerSpeedMult};
+
```

Comment:
With my proposed design, there's also no need to move the initialization of player after `onLoad`